feat(client): tsx compilation

a2937 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or Ona.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
I completely missed the fact that we needed to use Babel to separately transform Typescript React (TSX) files as we had Typescript and both React separately loaded. This lets it compile. 

When testing this on my computer (though the reloading made things weird); the syntax highlighting seemed to give off the exact same errors I would have if I were using pure Typescript. At the current moment of this PR, I'm pretty sure it's not a fluke. 

 By all counts, I'm not sure how to add the React library to the editor so that it gives similar React syntax highlighting like this lab. 

 https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/blocks/lab-color-picker/67bf4350777ac6ffdc027745.md

a2937 committed:
feat(tools): tsx compilation

a2937 commented:
<img width="1460" height="739" alt="Screenshot of the TSX editor so far" src="https://github.com/user-attachments/assets/15118072-c7cb-4d91-a435-997d203e3ef4" />


I started developing this on another fork but I believe all the relevant files are present. I also know the tests run but I think there's something off here. 


Like I think the TSXModuleTranspiler can't handle `presetsTSX` and prefers `presetsJSX` for some reason. 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/templates/Challenges/rechallenge/transformers.js#L195

<img width="1520" height="576" alt="image" src="https://github.com/user-attachments/assets/d280b4a4-7cd1-4891-a010-30043324de34" />

shootermv commented:
@a2937 , will it help you if i will test this PR on my machine?

a2937 commented:
I don't think so? I already have some code to test with.

shootermv commented:
i looked at edoitor of some react workshops:
here is what i see there (warnings at console  panel):
<img width="1310" height="559" alt="image" src="https://github.com/user-attachments/assets/de67b7e1-8d0e-4a18-a5b0-f2e13099ccb5" />

a2937 committed:
update webpack name

a2937 committed:
additional preset override

a2937 committed:
try to add react to typescript worker

a2937 committed:
revoke presetsTSX

a2937 committed:
use color picker app as temp challenge

raisedadead commented on curriculum/challenges/english/blocks/lab-color-picker/67bf4350777ac6ffdc027745.md:
```diff
@@ -232,7 +232,7 @@ export const ColorPicker = () => {};
+      src="index.tsx"
     ></script>
     <link rel="stylesheet" href="styles.css" />
 </head>
```
Line 226

a2937 commented:
Please see if the color picker challenge works for yourself. If it does, can this PR be merged and have the rough edges of this feature smoothed out at a later date.

ojeytonwilliams committed:
refactor: remove unnecessary babel changes

ojeytonwilliams committed:
feat: use React types in worker

ojeytonwilliams committed:
feat: bump ts, handle TSX, include global React

ojeytonwilliams committed:
feat: hard-code react types

a2937 commented:
Please do.

a2937 commented:
I'm still not entirely what changes are needed but please make them.

ojeytonwilliams committed:
fix: wait for ts worker before requesting compilation

ojeytonwilliams committed:
fix: dispose of models after use

ojeytonwilliams committed:
fix: only import monaco-editor types

Otherwise some code that uses window will be imported and this breaks
the build.

ojeytonwilliams committed:
fix: use allowUmdGlobalAccess so TS accepts global React

jdwilkin4 reviewed (changes requested):
It looks like all that is left is to remove the changes made to the color picker app. Then this can be merged in 👍🏾

a2937 committed:
undo changes to color picker lab
